### PR TITLE
🔒 : credentials mgmt

### PR DESCRIPTION
**Pull request by juwit**
This is a first attempt in implementing credentials management.

* The credentials are stored unencrypted in database.
* The credentials are injected as environment variables in the terraform container, when running the job.
* We try to avoid credential leaks through APIs (Stacks/Jobs).
* The credentials are scoped to a single user as for now.

Next steps to implement :
* scope credentials to a team
* encrypt and decrypt credentials using Vault (still storing them in database)
* get credentials in vault (only store a reference in database)

### CODE REVIEW

File: src/main/client/app/pages/credentials/credentials-routes.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const credentialsRoutes = [
+  {
+    path: '/credentials',
+    name: 'credentialsList',
+    component: () => import(/* webpackChunkName: "chunk-stacks" */ '@/pages/credentials/credentials-list.vue'),
```
Comment (cdubuisson) [inline src/main/client/app/pages/credentials/credentials-routes.js:5]: Shouldn't it have a specific chunk for lazy loading ?

File: src/main/client/app/pages/credentials/credentials-routes.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    meta: {
+      authorities: ['ROLE_USER'],
+      breadcrumb: [{ text: 'Credentials' }],
+      title: 'Gaia - Credentials',
+    },
+  },
+  {
+    path: '/credentials/new',
+    name: 'new_credentials',
+    component: () => import('@/pages/credentials/new-credentials'),
```
Comment (cdubuisson) [inline src/main/client/app/pages/credentials/credentials-routes.js:15]: same here

File: src/main/client/app/pages/credentials/credentials-routes.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    meta: {
+      authorities: ['ROLE_USER'],
+      breadcrumb: [{ text: 'Credentials', to: { name: 'credentialsList' } }, { text: 'New Credentials' }],
+      title: 'Gaia - New credentials edition',
+    },
+  },
+  {
+    path: '/credentials/:credentialsId',
+    name: 'credentials',
+    component: () => import(/* webpackChunkName: "chunk-modules" */ '@/pages/credentials/credentials.vue'),
```
Comment (cdubuisson) [inline src/main/client/app/pages/credentials/credentials-routes.js:25]: same here

File: src/main/client/app/pages/credentials/new-credentials.vue [hunk 4]
```diff
@@ -0,0 +1,87 @@
+<template>
+  <div>
+    <div v-if="credentials.provider == null">
+      <h1>Choose Terraform provider</h1>
+      <b-card-group columns>
+        <a @click="credentials.provider = 'aws'">
```
Comment (cdubuisson) [inline src/main/client/app/pages/credentials/new-credentials.vue:6]: cannot be a forEach with a list of providers ?

File: src/main/client/app/pages/credentials/providers/credentials-aws.vue [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    methods: {
+      notEmpty(field) {
+        return typeof field !== 'undefined' && field !== null && field.trim() !== '';
+      },
+    },
+  };
+</script>
+
+<style scoped>
```
Comment (cdubuisson) [inline src/main/client/app/pages/credentials/providers/credentials-aws.vue:47]: i think we can remove empty tags

File: src/main/client/app/pages/credentials/providers/credentials-azurerm.vue [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    methods: {
+      notEmpty(field) {
+        return typeof field !== 'undefined' && field !== null && field.trim() !== '';
+      },
+    },
+  };
+</script>
+
+<style scoped>
```
Comment (cdubuisson) [inline src/main/client/app/pages/credentials/providers/credentials-azurerm.vue:47]: i think we can remove empty tags

File: src/main/client/app/pages/credentials/providers/credentials-google.vue [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    methods: {
+      notEmpty(field) {
+        return typeof field !== 'undefined' && field !== null && field.trim() !== '';
+      },
+    },
+  };
+</script>
+
+<style scoped>
```
Comment (cdubuisson) [inline src/main/client/app/pages/credentials/providers/credentials-google.vue:40]: i think we can remove empty tags

**Comment by cdubuisson** [comment]
Just wondering if for components `credentials-aws`, `credentials-azurem`and `credentials-google`, it would be possible or not to create one component with dynamic form :thinking:

**Comment by juwit** [comment]
> Just wondering if for components `credentials-aws`, `credentials-azurem`and `credentials-google`, it would be possible or not to create one component with dynamic form thinking

I never liked the idea of dynamic forms. I always thought it was too complicated, and to hard to make it truly generic. So much special cases to handle (with rules for validation, and so on...).

**Pull request merged**